seacrowd/sea_datasets/id_msvd/id_msvd.py

```diff
@@ -0,0 +1,136 @@
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=f"{_DATASETNAME}",
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_{SEACROWD_SCHEMA_NAME}",
+            version=SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema=f"seacrowd_{SEACROWD_SCHEMA_NAME}",
+            subset_id=f"{_DATASETNAME}",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "video_path": datasets.Value("string"),
+                    "text": datasets.Sequence(datasets.Value("string")),
+                }
+            )
+        elif self.config.schema == f"seacrowd_{self.SEACROWD_SCHEMA_NAME}":
+            features = SCHEMA_TO_FEATURES[self.SEACROWD_SCHEMA_NAME.upper()]
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        # expect several minutes to download video data ~1.7GB
+        data_path = dl_manager.download_and_extract(_URLS)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "text_path": Path(data_path["text"]),
+                    "video_path": Path(data_path["video"]) / "YouTubeClips",
+                    "split": "train",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, text_path: Path, video_path:Path, split: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+        text_data = []
+        with open(text_path, "r", encoding="utf-8") as f:
+            for line in f:
+                id = line.find(" ")
+                video = line[:id]
+                text = line[id+1:].strip()
+                text_data.append([video, text])
+
+        df = pd.DataFrame(text_data, columns=["video_path", "text"])
+        df = df.groupby("video_path").agg(list).reset_index()
+        df['video_path'] = df['video_path'].apply(lambda x: video_path / f"{x}.avi")
+
+        if self.config.schema == "source":
+            for i, row in df.iterrows():
+                yield i, {
+                    "video_path": str(row["video_path"]),
+                    "text": row["text"],
+                }
+        elif self.config.schema == f"seacrowd_{self.SEACROWD_SCHEMA_NAME}":
+            for i, row in df.iterrows():
+                yield i, {
+                    "id": str(i),
+                    "video_path": str(row["video_path"]),
+                    "texts": row["text"],
```

Comment:
Furthermore, for the context of a Retrieval System, the data is constructed by creating a pair of query-document with a label (positive or negative). In this context a pair can be constructed by a video and a text; imagine searching a video on YT based on its video captioning rather than its title.

Therefore, the schema for VT2R should be extended to accommodate the label for a given pair (although it's all positive pairs in this dataset).

cc @holylovenia @SamuelCahyawijaya

Reply:
I see, then there will be several pairs for a video then.
I mistakenly assumed that `texts` field in video schema is a list, thanks for pointing that out.

I'm not sure about the general application of a Retrieval System in NLP. In video-text retrieval, the goal is mainly to find the most relevant video related to the semantics in a given sentence, and vice versa - with no text label in general. If we want to accommodate something (including labeling such as domain or video categories, etc), we can add `context` field as a child in `metadata`.

Reply:
Hi @akhdanfadh and @sabilmakbar, sorry I missed this discussion. We can use the `video` schema for this. The `texts` field does not necessarily refer to transcribed utterances, it could also refer to relevant queries as you said.

Reply:
8d8c04a : Changed the loading scheme based on the above, but I left the source as is. Also, adding a minor change to the key name in the `video` schema for clarity.

Reply:
cool, thanks and welcome back, @akhdanfadh! at a glance it looks okay, I'll test your newly implemented dataloader tonight. One thing, mind separating the video schema minor changes into a different PR? thanks in adv!